src/lib.rs

```diff
@@ -19,6 +19,8 @@
 
 extern crate alloc;
 
+// Required wildcard re-exporting to make all items available.
+// Should not short-circuit the "privacy chain".
 pub use ac_compose_macros::*;
 pub use ac_node_api::*;
 pub use ac_primitives::*;
```

Comment:
I do believe, that if we do not re-export these crates, we will have to add them individually over Cargo.   
Yes, we have here a problem with "general" type names, like Event,Error, ... because by re-exporting these base modules, we lose the specific module name in the path.

Reply:
My bad, we can use `pub extern crate`

Reply:
Ah, sorry I misformulated this. I think we can re-export this, but only the module itself:

```rust
pub use ac_compose_macros;
pub use ac_node_api;
pub use ac_primitives;
```

hmm, maybe we can rename this:
```rust
pub use ac_compose_macros as macros;
pub use ac_node_api as node_api;
pub use ac_primitives as primitives;
```

I am unsure though.